### PR TITLE
fixed links and some typos

### PR DESCRIPTION
**Pull request by mairaw**
while fixing some broken links from the build report, I also found some issues with the new topics.

I'd also consider adding Previous and Next buttons if they're a series. Not doing that on this PR.

/cc @rpetrusha @stevehoag

### CODE REVIEW

File: docs/csharp/getting-started/publishing-with-visual-studio-2017.md [hunk 1]
```diff
@@ -15,21 +15,20 @@ ms.assetid: a19545d3-24af-4a32-9778-cfb5ae938287
...
 Publishing .NET Core console applications from Visual Studio has not yet been implemented in the RC release. It will be implemented in the final release of Visual Studio 2017.
 
-You can publish your application using the [dotnet.exe](../../core/tools/dotnet.md) command line utility. To do this:
+You can publish your application using the [dotnet.exe](../../core/tools/dotnet.md) command-line utility. To do this:
 
 1. Open a console window. For example, in the **Ask me anything** text box in the Windows taskbar, enter `Command Prompt`, and then choose the **Command Prompt** desktop app to open the console window.
 
 1. Navigate to your library's project directory. Typically, unless you've reconfigured the file location, it is in the `Documents\Visual Studio 2017\Projects\HelloWorld\HelloWorld` directory. The directory contains your source code and a project file, `HelloWorld.csproj`.
 
-<!-- Steve, can you check this on your system? I'm sure I didn't do any renaming, but that's the directory I have on minem and I'd expect it to be HelloWorld instead of Hello. -->
```
Comment (mairaw) [inline docs/csharp/getting-started/publishing-with-visual-studio-2017.md]: I'm removing this comment but is there any action item left here? @stevehoag @rpetrusha

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md) | :warning:Warning | | [Details](#user-content-docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
[docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0415.md)
[docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0625.md)
[docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md) | :warning:Warning | | [Details](#user-content-docs/csharp/programming-guide/concepts/collections.md)
[docs/csharp/getting-started/publishing-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/publishing-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/publishing-with-visual-studio-2017.md)
[docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md)

<a name="docs/csharp/getting-started/debugging-with-visual-studio-2017.md"></a>
### [docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System.Console.ReadLine()&quot; sourcefile=&quot;docs/csharp/getting-started/debugging-with-visual-studio-2017.md&quot; sourcestartlinenumber=&quot;122&quot; sourceendlinenumber=&quot;122&quot;&gt;Console.ReadLine&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0415.md"></a>
### [docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime.CompilerServices.IndexerNameAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0415.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;IndexerNameAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0625.md"></a>
### [docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.FieldOffsetAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;FieldOffset&lt;/a&gt;&quot;, &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.StructLayoutAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;StructLayoutAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/programming-guide/concepts/collections.md"></a>
### [docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md)
 - **Line 214**: **[Warning]** Invalid file link:(~/docs/standard/collections/selecting-a-collection-class).
 - **Line 470**: **[Warning]** Invalid file link:(~/docs/csharp/programming-guide/concepts/BKMK_KindsOfCollections).



For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C17%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611170243110529-1259%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0415.md)
[docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0625.md)
[docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md) | :warning:Warning | | [Details](#user-content-docs/csharp/programming-guide/concepts/collections.md)
[docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
[docs/csharp/getting-started/publishing-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/publishing-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/publishing-with-visual-studio-2017.md)
[docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md)

<a name="docs/csharp/misc/cs0415.md"></a>
### [docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime.CompilerServices.IndexerNameAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0415.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;IndexerNameAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0625.md"></a>
### [docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.FieldOffsetAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;FieldOffset&lt;/a&gt;&quot;, &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.StructLayoutAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;StructLayoutAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/programming-guide/concepts/collections.md"></a>
### [docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md)
 - **Line Unspecified**: **[Warning]** Invalid file link:(~/docs/csharp/programming-guide/concepts/System.Collections namespaces.xml).




For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C17%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611170332485428-1259%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0415.md)
[docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0625.md)
[docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md) | :warning:Warning | | [Details](#user-content-docs/csharp/programming-guide/concepts/collections.md)
[docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
[docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md)

<a name="docs/csharp/misc/cs0415.md"></a>
### [docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime.CompilerServices.IndexerNameAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0415.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;IndexerNameAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0625.md"></a>
### [docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.FieldOffsetAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;FieldOffset&lt;/a&gt;&quot;, &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.StructLayoutAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;StructLayoutAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/programming-guide/concepts/collections.md"></a>
### [docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md)
 - **Line Unspecified**: **[Warning]** Invalid file link:(~/docs/csharp/programming-guide/concepts/System.Collections namespaces.xml).



For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C17%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611170357580641-1259%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0415.md)
[docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md) | :warning:Warning | | [Details](#user-content-docs/csharp/misc/cs0625.md)
[docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md) | :warning:Warning | | [Details](#user-content-docs/csharp/programming-guide/concepts/collections.md)
[docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
[docs/csharp/getting-started/publishing-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/publishing-with-visual-studio-2017.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/csharp/getting-started/publishing-with-visual-studio-2017.md)
[docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md) | :white_check_mark:Succeeded | | [Details](#user-content-docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md)

<a name="docs/csharp/misc/cs0415.md"></a>
### [docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime.CompilerServices.IndexerNameAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0415.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;IndexerNameAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0625.md"></a>
### [docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.FieldOffsetAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;FieldOffset&lt;/a&gt;&quot;, &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.StructLayoutAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;StructLayoutAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/programming-guide/concepts/collections.md"></a>
### [docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md)
 - **Line Unspecified**: **[Warning]** Invalid file link:(~/docs/csharp/programming-guide/concepts/System.Collections namespaces.xml).




For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C17%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611170405333239-1259%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/misc/cs0415?branch=pr-en-us-1259) | [Details](#user-content-docs/csharp/misc/cs0415.md)
[docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/misc/cs0625?branch=pr-en-us-1259) | [Details](#user-content-docs/csharp/misc/cs0625.md)
[docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/programming-guide/concepts/collections?branch=pr-en-us-1259) | [Details](#user-content-docs/csharp/programming-guide/concepts/collections.md)
[docs/csharp/getting-started/debugging-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/debugging-with-visual-studio-2017.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/debugging-with-visual-studio-2017?branch=pr-en-us-1259) | [Details](#user-content-docs/csharp/getting-started/debugging-with-visual-studio-2017.md)
[docs/csharp/getting-started/publishing-with-visual-studio-2017.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/getting-started/publishing-with-visual-studio-2017.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/publishing-with-visual-studio-2017?branch=pr-en-us-1259) | [Details](#user-content-docs/csharp/getting-started/publishing-with-visual-studio-2017.md)
[docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning?branch=pr-en-us-1259) | [Details](#user-content-docs/visual-basic/language-reference/error-messages/elementname-is-obsolete-visual-basic-warning.md)

<a name="docs/csharp/misc/cs0415.md"></a>
### [docs/csharp/misc/cs0415.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0415.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime.CompilerServices.IndexerNameAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0415.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;IndexerNameAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/misc/cs0625.md"></a>
### [docs/csharp/misc/cs0625.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/misc/cs0625.md)
 - **Line Unspecified**: **[Warning]** Invalid cross reference &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.FieldOffsetAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;FieldOffset&lt;/a&gt;&quot;, &quot;&lt;a href=&quot;xref:System​.Runtime​.InteropServices.StructLayoutAttribute&quot; sourcefile=&quot;docs/csharp/misc/cs0625.md&quot; sourcestartlinenumber=&quot;42&quot; sourceendlinenumber=&quot;42&quot;&gt;StructLayoutAttribute Class&lt;/a&gt;&quot;.

<a name="docs/csharp/programming-guide/concepts/collections.md"></a>
### [docs/csharp/programming-guide/concepts/collections.md](https://github.com/mairaw/docs/blob/brokenlinks/docs/csharp/programming-guide/concepts/collections.md)
 - **Line Unspecified**: **[Warning]** Invalid file link:(~/docs/csharp/programming-guide/concepts/System.Collections namespaces.xml).




For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C17%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611170559353116-1259%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mairaw** [comment]
I'm merging these fixes so I push changes into production now.

**Pull request merged**